### PR TITLE
GEOMESA-1494 HBase, Bigtable command-line tools

### PR DESCRIPTION
**Pull request by elahrvivaz**
~~note: stacks on top of #1178~~

 * AuditProvider moved to geomesa-utils to separate from Accumulo jars
* Fixed HBaseDataStore delete schema methods
* Moved scala 2.10 build script into build/ folder

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-accumulo/geomesa-accumulo-jobs/src/main/resources/org/locationtech/geomesa/jobs/accumulo-libjars.list [hunk 1]
```diff
@@ -1,14 +1,9 @@
-accumulo-core
-accumulo-fate
-accumulo-start
-accumulo-trace
+accumulo
```
Comment (jahhulbert-ccri) [inline geomesa-accumulo/geomesa-accumulo-jobs/src/main/resources/org/locationtech/geomesa/jobs/accumulo-libjars.list]: make sure we don't get any ```accumulo-server``` jar action up in there :smile:

File: geomesa-accumulo/geomesa-accumulo-jobs/src/test/scala/org/locationtech/geomesa/jobs/accumulo/AccumuloJobUtilsTest.scala [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import org.specs2.runner.JUnitRunner
+
+@RunWith(classOf[JUnitRunner])
+class AccumuloJobUtilsTest extends Specification {
+
+  "AccumuloJobUtils" should {
+    "load list of jars from class resource" in {
+      AccumuloJobUtils.defaultLibJars must not(beNull)
+      AccumuloJobUtils.defaultLibJars must not(beEmpty)
+      AccumuloJobUtils.defaultLibJars must contain("accumulo")
```
Comment (jahhulbert-ccri) [inline geomesa-accumulo/geomesa-accumulo-jobs/src/test/scala/org/locationtech/geomesa/jobs/accumulo/AccumuloJobUtilsTest.scala]: might be worth throwing in libthrift as well

File: geomesa-hbase/geomesa-bigtable-dist/src/main/assemblies/component.xml [hunk 3]
```diff
@@ -2,8 +2,20 @@
 <component xmlns="http://maven.apache.org/plugins/maven-assembly-plugin/component/1.1.2"
            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xsi:schemaLocation="http://maven.apache.org/plugins/maven-assembly-plugin/component/1.1.2 http://maven.apache.org/xsd/component-1.1.2.xsd">
+
     <dependencySets>
         <dependencySet>
+            <outputDirectory>lib</outputDirectory>
+            <directoryMode>0755</directoryMode>
+            <fileMode>0755</fileMode>
```
Comment (jahhulbert-ccri) [inline geomesa-hbase/geomesa-bigtable-dist/src/main/assemblies/component.xml]: i think  we are using 0644 for flies in lib

File: geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapred/GeoMesaOutputFormat.scala [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    ds.dispose()
+
+    // set the datastore parameters so we can access them later
+    GeoMesaConfigurator.setDataStoreOutParams(job, dsParams)
+    GeoMesaConfigurator.setSerialization(job)
+  }
+}
+
+/**
+ * Output format that turns simple features into mutations and delegates to AccumuloOutputFormat
```
Comment (jahhulbert-ccri) [inline geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapred/GeoMesaOutputFormat.scala]: maybe change comment to not specifically call out just accumulo ... also in mapreduce package

File: geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapred/GeoMesaOutputFormat.scala [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+      next.getUserData.putAll(value.getUserData)
+      writer.write()
+      progress.progress()
+    } catch {
+      case e: Exception => logger.error(s"Error writing feature '${DataUtilities.encodeFeature(value)}'", e)
+    }
+  }
+
+  override def close(reporter: Reporter) = {
+    writerCache.values.foreach(_.close())
```
Comment (jahhulbert-ccri) [inline geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapred/GeoMesaOutputFormat.scala]: possibly use IOUtils.closeQuietly to ensure the dispose() is called

File: geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapreduce/GeoMesaOutputFormat.scala [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+      written.increment(1)
+    } catch {
+      case e: Exception =>
+        logger.error(s"Error writing feature '${DataUtilities.encodeFeature(value)}'", e)
+        failed.increment(1)
+    }
+  }
+
+  override def close(context: TaskAttemptContext) = {
+    writerCache.values.foreach(_.close())
```
Comment (jahhulbert-ccri) [inline geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapreduce/GeoMesaOutputFormat.scala]: same comment on closeQuietly

**Comment by jahhulbert-ccri** [comment]
We may want to add a ```dist``` folder to include the gs plugin as well like we do for accumulo

File: geomesa-accumulo/geomesa-accumulo-jobs/src/test/scala/org/locationtech/geomesa/jobs/accumulo/AccumuloJobUtilsTest.scala (hunk 2 above)
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-jobs/src/test/scala/org/locationtech/geomesa/jobs/accumulo/AccumuloJobUtilsTest.scala]: added

File: geomesa-hbase/geomesa-bigtable-dist/src/main/assemblies/component.xml (hunk 3 above)
Comment (elahrvivaz) [inline geomesa-hbase/geomesa-bigtable-dist/src/main/assemblies/component.xml]: fixed

File: geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapred/GeoMesaOutputFormat.scala (hunk 4 above)
Comment (elahrvivaz) [inline geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapred/GeoMesaOutputFormat.scala]: fixed

File: geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapred/GeoMesaOutputFormat.scala (hunk 5 above)
Comment (elahrvivaz) [inline geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapred/GeoMesaOutputFormat.scala]: fixed

File: geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapreduce/GeoMesaOutputFormat.scala (hunk 6 above)
Comment (elahrvivaz) [inline geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/mapreduce/GeoMesaOutputFormat.scala]: fixed

**Comment by elahrvivaz** [comment]
It should include the gs-plugins - let me double check

**Comment by elahrvivaz** [comment]
It does build the gs-plugins in dist, but you have to use the `hbase` and/or `bigtable` profiles to get them. I updated the commit message but forgot to update the PR message - that's fixed now.

**Pull request merged**